packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -221,11 +215,28 @@ class TimeInput extends React.Component {
       }
     }
 
+    let skipStateUpdate = true;
+    // Data Attribute `data-date-time-input` of date-time-picker is used to ensure state update happens only when props are updated in date-time-picker.
+    const dateTimePicker = 'data-date-time-input';
+    if (this.props[dateTimePicker]) {
+      const inputValue = (this.props.showSeconds) ? `${this.hourInput.value}:${this.minuteInput.value}:${this.secondInput.value}` : `${this.hourInput.value}:${this.minuteInput.value}`;
+      const propsValue = (this.props.showSeconds) ? `${hour}:${minute}:${second}` : `${hour}:${minute}`;
+      // Comparing partial time values on each update will result in incorrect comparisons of inputValue and propsValue. Hence validating propsValue and inputValue before comparison.
+      if (TimeUtil.validateTime(propsValue, this.props.showSeconds) && TimeUtil.validateTime(inputValue, this.props.showSeconds)) {
+        skipStateUpdate = (propsValue === inputValue);
```

Comment:
I am certain I could be missing something here but do the member variables `hourInput`, `minuteInput`, `secondInput` reflect the values stored as the state? If so, what are we _actually_ comparing here to check whether or not the component has updated? We compare the current props and previous props values in Line 231 (which sounds right), are we comparing the current state value against current props value here?

Reply:
yes those variables actually reflects state values and Now instead of comparing states with props now I have updated Date-Time-Picker to call forceUpdate when timeValue and displayed are not same. : 94624440855ecca7249f9c3cd35ae80cda292300.